pkg/reconciler/broker/broker.go

```diff
@@ -695,37 +696,50 @@ func (r *Reconciler) reconcileIngressSubscription(ctx context.Context, b *v1alph
 func (r *Reconciler) reconcileIngressSubscriptionCRD(ctx context.Context, b *v1alpha1.Broker, c *duckv1alpha1.Channelable, svc *corev1.Service) (*v1alpha1.Subscription, error) {
 	expected := resources.MakeSubscriptionCRD(b, c, svc)
 
-	sub, err := r.getIngressSubscription(ctx, b)
+	sub, err := r.subscriptionLister.Subscriptions(b.Namespace).Get(expected.Name)
 	// If the resource doesn't exist, we'll create it
 	if apierrs.IsNotFound(err) {
+		logging.FromContext(ctx).Info("Creating subscription")
 		sub, err = r.EventingClientSet.EventingV1alpha1().Subscriptions(expected.Namespace).Create(expected)
 		if err != nil {
+			r.Recorder.Eventf(b, corev1.EventTypeWarning, ingressSubscriptionCreateFailed, "Broker's subscription create failed: %v", err)
 			return nil, err
 		}
 		return sub, nil
 	} else if err != nil {
+		logging.FromContext(ctx).Error("Failed to get subscription", zap.Error(err))
+		r.Recorder.Eventf(b, corev1.EventTypeWarning, ingressSubscriptionGetFailed, "Getting the Broker's Subscription failed: %v", err)
+		return nil, err
+	} else if !metav1.IsControlledBy(sub, b) {
+		b.Status.MarkIngressSubscriptionNotOwned(sub)
+		return nil, fmt.Errorf("broker %q does not own subscription %q", b.Name, sub.Name)
+	} else if sub, err = r.reconcileExistingIngressSubscription(ctx, b, expected, sub); err != nil {
 		return nil, err
 	}
+	return sub, nil
+}
 
+func (r *Reconciler) reconcileExistingIngressSubscription(ctx context.Context, b *v1alpha1.Broker, expected, actual *v1alpha1.Subscription) (*v1alpha1.Subscription, error) {
 	// Update Subscription if it has changed. Ignore the generation.
-	expected.Spec.DeprecatedGeneration = sub.Spec.DeprecatedGeneration
-	if !equality.Semantic.DeepDerivative(expected.Spec, sub.Spec) {
-		// Given that spec.channel is immutable, we cannot just update the subscription. We delete
-		// it instead, and re-create it.
-		err = r.EventingClientSet.EventingV1alpha1().Subscriptions(sub.Namespace).Delete(sub.Name, &metav1.DeleteOptions{})
-		if err != nil {
-			logging.FromContext(ctx).Info("Cannot delete subscription", zap.Error(err))
-			r.Recorder.Eventf(b, corev1.EventTypeWarning, ingressSubscriptionDeleteFailed, "Delete Broker Ingress' subscription failed: %v", err)
-			return nil, err
-		}
-		sub, err = r.EventingClientSet.EventingV1alpha1().Subscriptions(expected.Namespace).Create(expected)
-		if err != nil {
-			logging.FromContext(ctx).Info("Cannot create subscription", zap.Error(err))
-			r.Recorder.Eventf(b, corev1.EventTypeWarning, ingressSubscriptionCreateFailed, "Create Broker Ingress' subscription failed: %v", err)
-			return nil, err
-		}
+	expected.Spec.DeprecatedGeneration = actual.Spec.DeprecatedGeneration
+	if equality.Semantic.DeepDerivative(expected.Spec, actual.Spec) {
+		return actual, nil
 	}
-	return sub, nil
+	// Given that spec.channel is immutable, we cannot just update the subscription. We delete
```

Comment:
I don't think I understand the question/problem. I think you are asking i, ii, or iii below. 

We already have reconcilers watching the existing Subscriptions (e.g. the Trigger reconciler).

* In the world before this PR, the correct thing happens if that Subscription needs to change. Namely the Subscription is deleted and recreated.
* In the world after this PR, the correct thing may not happen because:
    1. A change to the old Subscription will cause a reconcile of the Trigger (because of stale watches)?
    1. The new Subscription will not be watched correctly?
    1. The new Subscription will not be deleted and recreated correctly?

Reply:
Sorry for the obviously way obtuse comment. What I should've been asking, are there any cases where we'd be creating the Subscription with a different name? I assume not.

Reply:
The newly created Subscription will have a different name than the 0.7 Subscription. However, it will always be on a distinct Channel than the 0.7 Subscription, so we should not see any duplicate events from having two Subscriptions.